backend/src/main/scala/com/softwaremill/adopttapir/starter/content/ContentService.scala

```diff
@@ -0,0 +1,51 @@
+package com.softwaremill.adopttapir.starter.content
+
+import com.softwaremill.adopttapir.logging.FLogging
+import com.softwaremill.adopttapir.starter.StarterDetails
+import com.softwaremill.adopttapir.template.ProjectGenerator
+
+class ContentService(projectGenerator: ProjectGenerator) extends FLogging {
+
+  def generateContentTree(starterDetails: StarterDetails): Node = {
+    val projectName = starterDetails.projectName
+    val projectAsFilesWithPaths = projectGenerator.generate(starterDetails)
+    val projectAsDirTrees = projectAsFilesWithPaths.map(gf => {
+      val pathsWithRootDir = (projectName + "/" + gf.relativePath).split('/').toList
+      val content = gf.content
+      createTree(pathsWithRootDir, content)
+    }).map(_.asInstanceOf[Directory])
```

Comment:
why would you need that cast?

Reply:
ah for the merge ... but what if we have top-level files? would be much better if this worked regardless of the file structure

Reply:
Since top level files are not allowed, it would throw an exception - but sure, I'll try to make it more explicit and reduce number of casts.